xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp

```diff
@@ -980,23 +980,22 @@ DemuxPacket* CDVDDemuxFFmpeg::Read()
       m_pkt.pkt.data = NULL;
 
       // timeout reads after 100ms
-      m_timeout.Set(20000);
+      m_timeout.Set(100);
       m_pkt.result = av_read_frame(m_pFormatContext, &m_pkt.pkt);
       m_timeout.SetInfinite();
     }
 
-    if (m_pkt.result == AVERROR(EINTR) || m_pkt.result == AVERROR(EAGAIN))
+    if (m_pkt.result == AVERROR(EINTR) || m_pkt.result == AVERROR(EAGAIN) ||
+        m_pkt.result == AVERROR_INVALIDDATA || m_pkt.result == AVERROR_EXIT)
     {
       // timeout, probably no real error, return empty packet
       bReturnEmpty = true;
     }
-    else if (m_pkt.result == AVERROR_EOF)
```

Comment:
When you change code you should always think of all use cases, not just your own you are trying to fix. If you drop this block, VP won't stop at the end of normal files. The code should like like this:

```
      m_timeout.Set(1000);
      m_pkt.result = av_read_frame(m_pFormatContext, &m_pkt.pkt);
      m_timeout.SetInfinite();
    }

    if (m_pkt.result == AVERROR(EINTR) || m_pkt.result == AVERROR(EAGAIN))
    {
      // timeout, probably no real error, return empty packet
      bReturnEmpty = true;
    }
    else if (m_pkt.result == AVERROR_EOF)
    {
      if (m_pInput->IsRealtime())
        bReturnEmpty = true;
    }
    else if (m_pkt.result < 0)
    {
      Flush();
      if (m_pInput->IsRealtime())
        bReturnEmpty = true;
    }
```

Reply:
yeah sorry that was an accident - I didn't mean to remove the AVERROR_EOF code.
will push your version soon.

Reply:
Just to clarify with the posted livestream I had issues on ts segment changes of the HLS stream when I used 100ms as timeout.
So I went with 500ms which seems to work fine, thought it's worth mentioning.

Reply:
I won't set timeout to less than 1000ms. If there is an issue, it does not matter if you notice it after 500ms or 1000ms. But if you run into timeout without an actual issue, you'll get video corruption because ffmpeg does not like truncated reads.

Reply:
Fair enough, thx for the info!